### PR TITLE
Create Redux slices: addresses & confirmed transactions

### PR DESCRIPTION
**Pull request by nop33**
To start working on integrating tokens, I want to make sure I can store them in Redux and not in our legacy contexts. To do that, I had to start moving away from the `addresses` context and into the `addressesSlice`. This PR is some necessary changes I needed to make so that I can later on start working on tokens. I had to stop making changes to keep the PR in reasonable size. I've added multiple TODOs. I will then work on pending transactions and if there's enough space, tokens.

- Duplicated client outside of the global context (will need to remove the one in the global context once all dependencies from it are removed)
- Started the `addressesSlice` and `confirmedTransactionsSlice` and added data to them when:
   - creating a new wallet
   - restoring a wallet
   - unlocking a wallet
   - fetching next transactions page of all addresses from Transfers page
   - fetching next transactions page of a specific address from the Address Detail page

### CODE REVIEW

**Comment by mvaivre** [comment]
👏 as always.

Functional tests, couple of things I noticed:

- I've sent a TX which was pending for 4min, then suddenly disappeared from the latest transactions list. Triggering "refresh data" wouldn't make it appear. I had to reload the app to see the TX.
- During the initial fetch, we're showing "no transaction" after the loading stopped, before actually displaying the TX list:

https://user-images.githubusercontent.com/6715943/217319067-fc967a33-35c5-4427-87af-c60125b5322d.mov

**Review by mvaivre — COMMENTED** [review]
Great job 👏

**Comment by nop33** [comment]
> Functional tests, couple of things I noticed:
>
> I've sent a TX which was pending for 4min, then suddenly disappeared from the latest transactions list. Triggering "refresh data" wouldn't make it appear. I had to reload the app to see the TX. On the token-2 branch, the pending transaction doesn't appear at all (will comment on the other PR as well).
During the initial fetch, we're showing "no transaction" after the loading stopped, before actually displaying the TX list: This seems to be fixed in token-2

Yes, this is because not everything is hooked up fully yet, some parts of the app are hooked on Redux, some are still on the context state. I thought it'd be a waste of time to make both work properly until the full migration happens and just went full speed ahead of finishing the migration. I'd propose that we do these functional tests once the state has moved to Redux. WDYT?

File: src/persistent-storage/address-metadata.ts [hunk 1]
```diff
@@ -16,16 +16,23 @@ You should have received a copy of the GNU Lesser General Public License
...
 */
 
-import { AddressMetadata, AddressSettings } from '@/types/addresses'
+import { AddressMetadata, AddressSettingsRedux } from '@/types/addresses'
 import { latestAddressMetadataVersion } from '@/utils/migration'
 
 import { DataKey, PersistentEncryptedStorage } from './encrypted-storage'
 
+interface AddressMetadataStorageStoreProps {
+  dataKey: DataKey
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/persistent-storage/address-metadata.ts:25]: > If I get it right, this is basically used to identify the right entry in the local storage and decrypt it. So in this case you mean "key" as a "key to unlock something"? Not a key like key: value, right?

Yes.

> WDYT? If this makes sense to you, let's continue the discussion in a separated issue.

Makes total sense, I'll create the issue and continue the discussion: https://github.com/alephium/desktop-wallet/issues/513

File: src/App.tsx [hunk 2]
```diff
@@ -51,6 +63,54 @@ const App = () => {
...
+        duration: 4000
+      })
+    } catch (e) {
+      dispatch(apiClientInitFailed())
+      console.error('Could not connect to network: ', network.name)
+      console.error(e)
+    }
+  }, [network.settings.nodeHost, network.settings.explorerApiHost, network.name, dispatch, setSnackbarMessage, t])
+
+  // Is there a better way to trigger the initial client initialization?
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/App.tsx:82]: I've been thinking about this. I personally feel that we are pressured with time to release the v2.0 as close to the Leman upgrade as possible. The current method of client initialization works and has been tested by time. I don't think it's the best way to do it, but it works! I think it's smarted to allocate our resources into bringing the v2.0 and the mobile wallet to life, and then optimize on React best practices. WDYT?

**Comment by mvaivre** [comment]
> I thought it'd be a waste of time to make both work properly until the full migration happens and just went full speed ahead of finishing the migration. I'd propose that we do these functional tests once the state has moved to Redux. WDYT?

100%, makes sense. I'll focus on code review until you give me your signal to start functional testing! 👌

File: src/App.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvaivre) [reply src/App.tsx:82]: Alright, let's keep that in mind. I've created [an issue](https://github.com/alephium/desktop-wallet/issues/514) about it.

**Comment by nop33** [comment]
@mvaivre Nice! can I merge this then?

**Review by mvaivre — APPROVED** [review]

**Pull request merged**